internal/controllers/services.go

```diff
@@ -138,6 +141,70 @@ func (r *Reconciler) reconcileAgentService(ctx context.Context, cr *model.Cryost
 	})
 }
 
+func (r *Reconciler) newAgentHeadlessService(cr *model.CryostatInstance, namespace string) *corev1.Service {
+	return &corev1.Service{
```

Comment:
If I understand correctly, this Service is made headless by having a `ClusterIP: nil` in its spec. I guess here that is done implicitly, but would it be worth setting it explicitly for readability?